utils/api/utils.ts

```diff
@@ -28,7 +37,13 @@ export const pretreat = async <T>(res: Response) => {
   if (res.status === HttpStatus.NotFound) {
     throw new NotFoundException()
   }
+
   const parsed: T | ErrorResponse = await res.json()
+
+  if ('errors' in parsed && parsed.errors.status === ServerErrorCode.UseEthHash) {
```

Comment:
Why not use `parsed?.errors.status`

Reply:
`'errors' in parsed` is a type guard here to declare the type of `parsed` is `ErrorResponse`. Without this, the `errors` should be declared in `T` too

Reply:
Gotcha